system/Files/FileSizeUnit.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * This file is part of CodeIgniter 4 framework.
+ *
+ * (c) CodeIgniter Foundation <admin@codeigniter.com>
+ *
+ * For the full copyright and license information, please view
+ * the LICENSE file that was distributed with this source code.
+ */
+
+namespace CodeIgniter\Files;
+
+enum FileSizeUnit: int
+{
+    case B  = 0;
+    case KB = 1;
+    case MB = 2;
+    case GB = 3;
+    case TB = 4;
+}
```

Comment:
Wouldn't it be nice to have the option to select the desired unit with a string, as we have in the depreciated method? This will allow easier migration if the value is stored in a database.
```suggestion

    public static function fromString(string $unit): self
    {
        return match (strtolower($unit)) {
            'b'  => self::B,
            'kb' => self::KB,
            'mb' => self::MB,
            'gb' => self::GB,
            'tb' => self::TB,
            default => throw new InvalidArgumentException("Invalid unit: $unit"),
        };
    }
}
```